app/controllers/users_controller.rb

```diff
@@ -150,7 +152,8 @@ def update
       @user = User.find(params[:id])
       respond_to do |format|
         if @user.update_attributes(params[:user])
-          @user.set_role_ids(params[:user][:role_ids]) if params[:user][:role_ids]
+          @user.set_role_ids(params[:user][:role_ids] || [])
```

Comment:
This is the bug that was also fixed.  If the user had roles when user/edit loads and then all the role checkboxes are unchecked @user.set_roles_ids() would be skipped since params[:user][:role_ids] is empty and all the user's roles would not be removed.